cpp/src/gandiva/function_registry_string.cc

```diff
@@ -225,6 +225,67 @@ std::vector<NativeFunction> GetStringFunctionRegistry() {
                      "convert_replace_invalid_fromUTF8_binary",
                      NativeFunction::kNeedsContext),
 
+      NativeFunction("convert_todouble", {}, DataTypeVector{float64()}, binary(),
```

Comment:
nit: lets keep the names as "convert_toDOUBLE", reads better

Reply:
Should I need to apply that sugestion to other functions too? I am asking it because in this comment https://github.com/apache/arrow/pull/9861#discussion_r622908353 you said to not use case-sensitive names

Reply:
I said function_names are not case-sensitive. So you don't need to have both upper case and lower case variants, only one would do.

Reply:
Should fix it if some part of code assumes the names are case-sensitive.

Reply:
Ok, I understand now. I applied your suggestion.

Reply:
Suggestions applied and I checked and there's no part that assumes that names are case-sensitive